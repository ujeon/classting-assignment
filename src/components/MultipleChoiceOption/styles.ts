import colors from '@themes/colors';
import styled from 'styled-components/native';

interface OptionProps {
  selected: boolean;
}

export const OptionContainer = styled.TouchableOpacity`
  justify-content: center;
  align-items: center;
  width: 100%;
  margin-top: 10px;
`;

export const Option = styled.View<OptionProps>`
  width: 100%;
  min-height: 50px;
  background-color: ${({ selected }) => (selected ? colors.lightCyan : colors.white)};
  border-radius: 12px;
  border: 1px ${({ selected }) => (selected ? colors.lightSkyBlue : colors.gainsboro)};
  justify-content: center;
  padding: 5px 13px;
  z-index: 10;
`;

export const OptionShadow = styled.View<OptionProps>`
  width: 100%;
  min-height: 50px;
  background-color: ${({ selected }) => (selected ? colors.lightSkyBlue : colors.gainsboro)};
  position: absolute;
  border-radius: 12px;
  top: 3px;
`;

export const Contents = styled.Text<OptionProps>`
  color: ${({ selected }) => (selected ? colors.dodgerBlue : colors.darkGrey)};
`;
