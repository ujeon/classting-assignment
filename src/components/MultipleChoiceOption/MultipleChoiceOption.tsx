import React from 'react';
import { Contents, Option, OptionContainer, OptionShadow } from './styles';

interface MultipleChoiceOptionProps {
  selected: boolean;
  contents: string;
  onPress: () => void;
}

const MultipleChoiceOption: React.FC<MultipleChoiceOptionProps> = ({
  selected,
  contents,
  onPress,
}) => {
  return (
    <OptionContainer onPress={onPress}>
      <Option selected={selected}>
        <Contents selected={selected}>{contents}</Contents>
      </Option>
      <OptionShadow selected={selected} />
    </OptionContainer>
  );
};

export default MultipleChoiceOption;
