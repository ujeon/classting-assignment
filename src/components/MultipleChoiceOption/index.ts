import MultipleChoiceOption from './MultipleChoiceOption';

export default MultipleChoiceOption;
