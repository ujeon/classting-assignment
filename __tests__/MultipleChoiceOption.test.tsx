import React from 'react';
import renderer from 'react-test-renderer';
import MultipleChoiceOption from '@components/MultipleChoiceOption';

test('[MultipleChoiceOption] render:', () => {
  const tree = renderer
    .create(<MultipleChoiceOption selected onPress={() => {}} contents="테스트 문구" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
