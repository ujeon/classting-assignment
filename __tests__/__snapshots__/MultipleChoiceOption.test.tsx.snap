// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MultipleChoiceOption] render: 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 10,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    selected={true}
    style={
      Array [
        Object {
          "backgroundColor": "#DDF4FF",
          "borderColor": "#83D7FE",
          "borderRadius": 12,
          "borderStyle": "solid",
          "borderWidth": 1,
          "justifyContent": "center",
          "minHeight": 50,
          "paddingBottom": 5,
          "paddingLeft": 13,
          "paddingRight": 13,
          "paddingTop": 5,
          "width": "100%",
          "zIndex": 10,
        },
      ]
    }
  >
    <Text
      selected={true}
      style={
        Array [
          Object {
            "color": "#3CAFF5",
          },
        ]
      }
    >
      테스트 문구
    </Text>
  </View>
  <View
    selected={true}
    style={
      Array [
        Object {
          "backgroundColor": "#83D7FE",
          "borderRadius": 12,
          "minHeight": 50,
          "position": "absolute",
          "top": 3,
          "width": "100%",
        },
      ]
    }
  />
</View>
`;
